opentok/opentok.py

```diff
@@ -439,8 +454,108 @@ def get_archives(self, offset=None, count=None):
             return ArchiveList(self, response.json())
         elif response.status_code == 403:
             raise AuthError()
+        else:
+            raise RequestError("An unexpected error occurred", response.status_code)
+
+    def signal(self, session_id, connection_id, payload):
+        """Sends a signal to all the connections in a session or to a specific one.  This API is the
+          server side API equivalent to the signal method in the Client SDKs:
+          https://www.tokbox.com/developer/guides/signaling/js/
+
+        :param String session_id: The session_id where you want to send the signal to.
+        :param String connection_id: The connection_id of a connection in the session.   Leave
+          it empty if you want to send a signal to all the connections in the session.
+        :param Dictionary payload: Object with optional signal type and signal data fields.ope
+        """
+        if not payload.get('data'):
+            raise OpenTokException(u('Cannot send a signal without data property in the payload'))
+
```

Comment:
The docs say that `data` is optional (and I think it should be). I recommend removing this exception.